### PR TITLE
[MM-26482] Textbox Common Component for Cloud Onboarding

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
Textbox Common Component for Cloud Onboarding
Ported over from the `customer-web-server` repo and modified to spec.

Demo included based on current work done.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-26482

### CODE REVIEW

**Comment by devinbinnie** [comment]
> Thanks @devinbinnie. Here's my review:
> 
> 1. Can we animate the state transition from placeholder to focus? Here's an example animation in Figma: https://www.figma.com/proto/uv9A6dZsekQHHhOhYC7yc2/Cloud-UX?node-id=3133%3A34&viewport=376%2C526%2C0.3246636390686035&scaling=min-zoom

Done, I think I have it close but again Figma doesn't provide exact specs for animations. Let me know if it needs tweaking.

> 2. It looks like things jump around a little when you focus on the input field - I'm guessing that's because the border changes from 1px to 2px - I think Asaad had a solution to work around this - check with him maybe. This applies to the error focus state as well

Should be good to go now.

> 3. I'm going to assume any max-width would be placed on the container of the inputs, not the inputs themselves, so I won't bother calling the width out at this point

Yeah that's correct, this is just a demo so the `width` will be defined based on whatever component consumes this.

> 4. Error state: this displays the second I type something. We should reserve errors until users leave focus of the field - not to put it in error state as you're typing. But maybe this is temporary for testing?

This is just temporary for testing, you can control when the input shows an error. Will keep that in mind when designing the actual consumer component.

> 5. Help text/error text: should have line-height of 16px, and margin top of 8px

Done!

> 6. If the error text wraps, it should not wrap under the icon
>    image

Fixed!

> 7. There appears to be something odd with the rounded corners. They aren't perfectly rounded for some reason (only in the default state)
>    image

Fixed!

**Comment by devinbinnie** [comment]
@matthewbirtch Asaad assisted with some better CSS so things should be fixed now. Let me know if anything is out of place.

File: components/input.css [hunk 1]
```diff
@@ -0,0 +1,117 @@
+.Input {
+    width: 100%;
+    position: relative;
+    left: 0px;
+    border: 0;
+    outline: 0;
+    color: var(--center-channel-color);
+    /*margin-right: 15px;*/
```
[the remark this answers is not in the document]
Comment (devinbinnie) [reply components/input.css:8]: Yes!

File: components/input.css [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    color: rgba(var(--center-channel-color-rgb), 0.56);
+}
+
+.Input-info > span.valid {
+    margin-left: 7px;
+}
+
+.Input-fieldset {
+    width: 100%;
+    /*height: 48px;*/
```
[the remark this answers is not in the document]
Comment (devinbinnie) [reply components/input.css:60]: _Yes but italicized!_

**Comment by deanwhillier** [comment]
Having an offline conversation with @devinbinnie, will post here with any updates as needed.

**Pull request merged**